pkg/scanner/scanner.go

```diff
@@ -102,12 +102,62 @@ func NewScanner(resourceFetcher ResourceFetcher, logger Logger) *Scanner {
 	}
 }
 
+// ValidateRule validates a rule without executing it
+// This method allows SDK users to validate CEL expressions before deployment
+func (s *Scanner) ValidateRule(rule Rule) ValidationResult {
+	validator := NewRuleValidator(s.logger)
+	return validator.ValidateRule(rule)
+}
+
+// ValidateCELExpression validates a CEL expression with given inputs
+// This is a convenience method for validating just the expression
+func (s *Scanner) ValidateCELExpression(expression string, inputs []Input) error {
+	return CompileCELExpression(expression, inputs)
+}
+
+// ValidateAllRules validates all rules in a ScanConfig without executing them
+// Returns a map of rule ID to ValidationResult for detailed analysis
+func (s *Scanner) ValidateAllRules(config ScanConfig) map[string]ValidationResult {
```

Comment:
I imagine the idea here is to call this from the ComplianceScan controller to validate the all rules compile before kicking off the actual scan?